app/helpers/chart_helper.rb

```diff
@@ -1,4 +1,40 @@
 module ChartHelper
-  
-  
+  def pluralize(symbol)
+    return symbol.to_s.pluralize.to_sym
+  end
+  def stackedSubtypeChart(grouping)
+    rsubs = Measurement
+              .where("measurements.date >= ? AND measurements.date <= ?",
+                  params[:start].to_date.advance(pluralize(grouping) => -1),
+                  params[:stop].to_date.advance(pluralize(grouping) => 1))
+              .joins(sensor: {subtype: :type})
+              .where("types.resource = ?", @rname)
+              .order('subtypes."usage?"')
+              .group_by_period(grouping, :date).group("subtypes.id, subtypes.name").maximum(:amount)
+    agg = {}
+    rsubs.map do |(date, stype), value|
+      agg[stype] ||= { name: stype, data: [] }
+      agg[stype][:data].append([date,value])
+    end
+    rsubs = agg.values
+    rsubs.each_with_index do |stype, si|
+      stype[:data].to_enum.with_index.reverse_each do |v, i|

```

Comment:
Do you need `to_enum` here? It's already an array...

Reply:
If you implement my comment on ln17, you could just do: `stype[:data].each_with_index`
Also, if you were using `to_enum` to get each array value as its own argument to the block, you can decompose array arguments with parens (like I did on line 15):

`def foo(a, (b, c)) => foo(1, [2,3]) => 1, 2, 3`

Reply:
Also, as a note, idiomatic ruby favors descriptive, full english variable names.

Reply:
That might actually be worth having to reverse again at the end. will try that out.